src/finn/transformation/fpgadataflow/templates.py

```diff
@@ -126,120 +126,113 @@ def __init__(
         # used by TLastMarker to signal end of transmission for AXI CDMA
         self.REG_OFFSET_NUM_ITERS = 0x10
 
-    def load_input(self):
+    def load_input(self, inputfile):
         N = self.N
-        ishape_normal = $INPUT_SHAPE_NORMAL$
         # load desired input .npy file
-        ibuf_normal = np.load(self.inputfile)
+        ibuf_normal = np.load(inputfile)
         # ensure that shape is as expected
-        assert ibuf_normal.shape == ishape_normal
+        assert ibuf_normal.shape == self.ishape_normal
         return ibuf_normal
 
     def pack_input(self, ibuf_normal):
         N = self.N
-        # input FINN DataType
-        idt = $INPUT_FINN_DATATYPE$
-        ishape_folded = $INPUT_SHAPE_FOLDED$
         # convert to folded form
-        ibuf_folded = ibuf_normal.reshape(ishape_folded)
+        ibuf_folded = ibuf_normal.reshape(self.ishape_folded)
         # pack the input buffer, reversing both SIMD dim and endianness
         ibuf_packed = finnpy_to_packed_bytearray(
-            ibuf_folded, idt, reverse_endian=True, reverse_inner=True
+            ibuf_folded, self.idt, reverse_endian=True, reverse_inner=True
         )
         return ibuf_packed
 
     def unpack_output(self, obuf_packed):
         N = self.N
-        # output FINN DataType
-        odt = $OUTPUT_FINN_DATATYPE$
-        oshape_folded = $OUTPUT_SHAPE_FOLDED$
         # unpack the packed output buffer from accelerator
         obuf_folded = packed_bytearray_to_finnpy(
-            obuf_packed, odt, oshape_folded, reverse_endian=True, reverse_inner=True
+            obuf_packed, self.odt, self.oshape_folded, reverse_endian=True, reverse_inner=True
         )
         return obuf_folded
 
-    def save_output(self, obuf_folded):
+    def save_output(self, obuf_folded, outputfile):
```

Comment:
similar to above wrt load_input
move unfolding of output to own member function and get rid of save_output, user can call np.save themselves